### PR TITLE
fix: add 3 tries for socket read

### PR DESCRIPTION
**Pull request by RomanAyupov89**

### CODE REVIEW

File: src/marionette/transport.cr [hunk 1]
```diff
@@ -71,11 +72,24 @@ module Marionette
       Message.new(type.as_i, id.as_i, command.as_s?, params)
     end
 
+    private def get_string
+      3.times do |i|
+        s = @socket.gets(':')
+        Log.trace { "(#{@id}) Marionette receive message id: #{@last_id}, timeout #{@timeout} after #{i} tries" }
```
Comment (alonchin) [inline src/marionette/transport.cr]: we don't have timeout here, right?
```suggestion
        Log.trace { "(#{@id}) Marionette received message id: #{@last_id} after #{i} tries" }
```

File: src/marionette/transport.cr [hunk 2]
```diff
@@ -71,11 +72,24 @@ module Marionette
...
+        return s
+      rescue Error::TimeoutError
+        Log.error { "(#{@id}) Marionette can not receive message id: #{@last_id}, timeout #{@timeout} after #{i} tries" }
+        sleep 1.seconds
+      end
+      raise "Error getting Marionette after 3 tries"
```
Comment (alonchin) [inline src/marionette/transport.cr]:
```suggestion
      raise "Error reading from the socket after 3 attempts"
```

File: src/marionette/transport.cr (hunk 1 above)
Comment (RomanAyupov89) [reply src/marionette/transport.cr]: why?@socket.read_timeout = @timeout ln 22

Comment (alonchin) [reply src/marionette/transport.cr]: it is a bit confusing. If we get a message from the socket we still have the log that message is received, but there was some timeout after 1st attempt.
We have TimeoutError exception handling to log messages about that.

**Review by alonchin — APPROVED** [review]

**Pull request merged**